### ISSUE TITLE
If lutron bridge disconnects this integration can block all i/o on home assistant

### ISSUE THREAD

**Issue by bdraco** [upsert/lutron-caseta-pro#50]
```
bash-5.0# ../py-spy top --pid 201 


Collecting samples from 'python3 -m homeassistant --config /config' (python v3.7.6)
Total Samples 100
GIL: 63.33%, Active: 474.44%, Threads: 61

  %Own   %Total  OwnTime  TotalTime  Function (filename:line)                                                                                                                                                                                                                                                 
190.00% 190.00%    1.75s     1.75s   _worker (concurrent/futures/thread.py:78)
 70.00%  70.00%   0.690s    0.690s   _read_until (lutron_caseta_pro/casetify.py:179)
 56.67%  56.67%   0.660s    0.660s   recv (pyhap/hap_server.py:725)
 37.78%  37.78%   0.360s    0.360s   handle_read (zeroconf/__init__.py:1265)
 34.44%  34.44%   0.320s    0.320s   run (zeroconf/__init__.py:1226)
 31.11%  31.11%   0.310s    0.310s   _loop (senseme/lib/background_monitor.py:34)

```

**Comment by bdraco** [comment]
<img width="797" alt="Screen Shot 2020-03-26 at 8 37 33 AM" src="https://user-images.githubusercontent.com/663432/77652962-18eb8d80-6f3d-11ea-816c-0b84c6ad3bcc.png">

**Comment by bdraco** [comment]
Maybe need to check for empty reads on this line and then reconnect?

https://github.com/upsert/lutron-caseta-pro/blob/master/custom_components/lutron_caseta_pro/casetify.py#L189

**Comment by bdraco** [comment]
```
    async def _read_until(self, value):
        """Read until a given value is reached."""
        while True:
            if hasattr(value, "search"):
                # assume regular expression
                match = value.search(self._read_buffer)
                if match:
                    self._read_buffer = self._read_buffer[match.end():]
                    return match
            else:
                where = self._read_buffer.find(value)
                if where != -1:
                    self._read_buffer = self._read_buffer[where + len(value):]
                    return True
            try:
                read = await self.reader.read(READ_SIZE)
                if not len(read):
                    _LOGGER.warning("Empty read from Lutron bridge (may be disconnected)")
                    return False
                self._read_buffer += read
            except OSError as err:
                _LOGGER.warning("Error reading from Lutron bridge: %s", err)
                return False
```
This seems to fix it but needs more testing

**Comment by bdraco** [comment]
Reconnect doesn't seem to be working for me.

I've been able to replicate this by unplugging the box at inopportune times

**Comment by bdraco** [comment]
Looks like the writes need to `await self.writer.drain()`

**Comment by bdraco** [comment]
Looks like the ping is holding the write lock forever when it disconnects maybe asynctimeout

**Comment by bdraco** [comment]
Unplug bridge and the ping is happily chugging along even though it can't write
```
2020-03-26 22:17:06 DEBUG (MainThread) [custom_components.lutron_caseta_pro.casetify] Start ping
2020-03-26 22:17:06 DEBUG (MainThread) [custom_components.lutron_caseta_pro.casetify] write ping


```

**Comment by cajuncoding** [comment]
Interesting find!  I'm pretty sure I've experienced this also before.... ended up with a full VM restart to get my CPU levels back to normal.

**Comment by bdraco** [comment]
I've got a working fix.  I'll send a PR shortly.

Testing:
Disconnect lutron bridge
Wait 2m
Reconnect lutron bridge
Wait 10m

Verify everything is working

**Comment by upsert** [comment]
Fixed by #51
